### PR TITLE
feat(TCOMP-1685): use tsbi java8-svc-base image

### PR DESCRIPTION
**Pull request by undx**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

File: images/component-server-image/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+             "--http=${boundPort}"
+             ]
+
+LABEL com.talend.name="Talend Component Kit Server"
+LABEL com.talend.application="component-server"
+LABEL com.talend.service="component-server"
+LABEL com.talend.description="Talend Component Kit Backend Server"
+LABEL com.talend.version="${PROJECT_VERSION}"
+LABEL com.talend.build-date="${TIMESTAMP}"
+LABEL com.talend.vendor="Talend"
```
Comment (asaintsever) [inline images/component-server-image/Dockerfile:57]:
```suggestion
```
Not necessary, already defined see: https://github.com/Talend/platform-services-microservices#labels

File: images/component-server-image/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+LABEL com.talend.maintainer="Talend <support@talend.com>"
```
Comment (asaintsever) [inline images/component-server-image/Dockerfile:58]:
```suggestion
```
Not necessary, already defined see: https://github.com/Talend/platform-services-microservices#labels

File: images/component-server-image/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+LABEL com.talend.url="https://www.talend.com"
```
Comment (asaintsever) [inline images/component-server-image/Dockerfile:59]:
```suggestion
```
Not necessary, already defined see: https://github.com/Talend/platform-services-microservices#labels

File: images/component-server-image/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+LABEL com.talend.git.branches="${git.branch}"
+LABEL com.talend.git.commits="${git.commit.id}"
+LABEL com.talend.docker.cmd="docker run -d -p ${boundPort}:${boundPort} tacokit/component-server:${PROJECT_VERSION}"
+LABEL com.talend.docker.params="_JAVA_OPTIONS=<JVM options (system properties etc), ex: -Dtalend.component.server.component.registry=/path/to/component-registry.propertes -Dtalend.component.server.maven.repository=/path/to/m2>"
+LABEL com.talend.docker.healthcheck="curl --fail http://localhost:${boundPort}/api/v1/environment"
+
+ENV MEECROWAVE_HOME       ${imageWorkDir}
+ENV MEECROWAVE_BASE       ${imageWorkDir}
+ENV MEECROWAVE_PID        ${imageWorkDir}/conf/server.pid
+ENV LANG                  en_US.UTF-8
```
Comment (asaintsever) [inline images/component-server-image/Dockerfile:70]:
```suggestion
```
Not necessary, already defined see: https://github.com/Talend/platform-services-microservices#environment-variables

File: images/component-server-image/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ENV LOGGING_LAYOUT        TEXT
```
Comment (asaintsever) [inline images/component-server-image/Dockerfile:71]: Are you sure to set it to `TEXT` ?

By default it is set to `JSON`: https://github.com/Talend/platform-services-microservices#environment-variables
It needs to be set to `JSON` if you want your logs to be collected on K8S. Also, logs must be sent to stdout, not in a file!

File: images/component-server-image/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ARG boundPort=8080
+#
+ARG TALEND_JDK_SERIAL_FILTER
+ARG LOGGING_LAYOUT
+
+FROM ${DOCKER_REGISTRY_HOST}${TSBI_IMAGE_PATH}/java8-svc-base:${TSBI_VERSION}
+
+EXPOSE ${boundPort}
+
+COPY ${ADDITIONAL_FILES}/* ${imageWorkDir}/
```
Comment (asaintsever) [inline images/component-server-image/Dockerfile:31]: What kind of files are you copying?

File: images/component-server-image/Dockerfile (hunk 1 above)
Comment (undx) [reply images/component-server-image/Dockerfile:57]: removed

File: images/component-server-image/Dockerfile (hunk 3 above)
Comment (undx) [reply images/component-server-image/Dockerfile:59]: removed

File: images/component-server-image/Dockerfile (hunk 4 above)
Comment (undx) [reply images/component-server-image/Dockerfile:70]: removed

File: images/component-server-image/Dockerfile (hunk 2 above)
Comment (undx) [reply images/component-server-image/Dockerfile:58]: removed

File: images/component-server-image/Dockerfile (hunk 5 above)
Comment (undx) [reply images/component-server-image/Dockerfile]: Changed.

File: images/component-server-image/Dockerfile (hunk 6 above)
Comment (undx) [reply images/component-server-image/Dockerfile]: some resources to a conf folder.

File: images/component-server-image/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+LABEL com.talend.version="$PROJECT_VERSION"
+LABEL com.talend.build-date="$TIMESTAMP"
+LABEL com.talend.docker.cmd="docker run -d -p ${BOUND_PORT}:${BOUND_PORT} tacokit/component-server:$PROJECT_VERSION"
+LABEL com.talend.docker.params="_JAVA_OPTIONS=<JVM options (system properties etc), ex: -Dtalend.component.server.component.registry=/path/to/component-registry.propertes -Dtalend.component.server.maven.repository=/path/to/m2>"
+LABEL com.talend.docker.healthcheck="curl --fail http://localhost:${BOUND_PORT}/api/v1/environment"
+LABEL com.talend.git.repositories="https://github.com/talend/component-runtime"
+LABEL com.talend.git.branches="${GIT_BRANCH}"
+LABEL com.talend.git.commits="${GIT_COMMIT}"
+
+COPY --chown=talend:talend target/classes/docker/additional/ ${TALEND_APP_HOME}/
```
[the remark this answers is not in the document]
Comment (undx) [reply images/component-server-image/Dockerfile:39]: There's no jars inside that folder, that's resources remaped to a conf folder and a dynamic library.

File: images/component-server-image/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+COPY --chown=talend:talend target/classes/org/               ${TALEND_APP_HOME}/classes/org/
```
[the remark this answers is not in the document]
Comment (undx) [reply images/component-server-image/Dockerfile:40]: > Do not understand why you need to copy that: TSBI already copied jar component-server-image-1.1.20-SNAPSHOT.jar under ${TALEND_APP_HOME}/lib/ and this jar contains classes/org/. So ?

You're right, I'll remove that one.

> I see that you specifically set -cp options in JAVA_OPTS env: the ${TALEND_APP_HOME}/lib/* part is basically what is achieved by what I mentioned above. Unless you had already checked that you do need to copy the classes + resources and define the classpath the way you did, maybe you could try what I suggest to end up with a simpler Dockerfile.

We have some extra classpath points for customization, so we keep the `-cp` option. I can effectively replace the `lib/*` by the env variable `CLASSPATH`.

File: images/component-server-image/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+ENV MEECROWAVE_HOME       ${TALEND_APP_HOME}
+ENV MEECROWAVE_BASE       ${TALEND_APP_HOME}
+ENV MEECROWAVE_PID        ${TALEND_APP_HOME}/conf/server.pid
+ENV LOGGING_LAYOUT        JSON
```
[the remark this answers is not in the document]
Comment (undx) [reply images/component-server-image/Dockerfile]: removed

**Pull request merged**